src/lib/server/services/users.ts

```diff
@@ -117,17 +117,34 @@ export const userIsBlocking = async (db: DB, userId: string, blockedUserId: stri
     return !!block;
 };
 
+export const blockedByUser = async (db: DB, userId: string): Promise<User[]> => {
+    const blocked = await db
+        .select()
+        .from(userBlockTable)
+        .innerJoin(userTable, eq(userBlockTable.blockedUserId, userTable.id))
+        .where(eq(userBlockTable.userId, userId));
+    return blocked.map((u) => u.user);
+};
+
 export const getUserByUsername = async (db: DB, username: string) => {
     const [user] = await db.select().from(userTable).where(eq(userTable.username, username));
     return user;
 };
 
+export const updateUser = async (
+    db: DB,
+    user: Pick<User, 'id'> & Omit<Partial<User>, 'username' | 'profileImage'>
```

Comment:
This is not correct.  It should be
```suggestion
    user: Pick<User, 'id'> & Pick<Partial<User>, 'username' | 'profileImage'>
```